astropy/units/physical.py

```diff
@@ -530,20 +530,23 @@ def get_physical_type(obj):
 
 # This generates a docstring addition for this module that describes all of the
 # standard physical types defined here.
+# TODO! combine the label with {name}
 if __doc__ is not None:
     doclines = [
         ".. list-table:: Defined Physical Types",
         "    :header-rows: 1",
-        "    :widths: 30 10 50",
+        "    :widths: 1 30 10 50",
         "",
-        "    * - Physical type",
+        "    * - ",
+        "      - Physical type",
         "      - Unit",
         "      - Other physical type(s) with same unit"]
 
     for name in sorted(_name_physical_mapping.keys()):
         physical_type = _name_physical_mapping[name]
         doclines.extend([
-            f"    * - {name}",
+            f"    * - .. _ptype-{name.replace(' ', '_')}: ",
+            f"      - {name}",
```

Comment:
Wouldn't it work to use a direct reference, as in `my custom text <target to text>`_? (or, in this context,
```
- `{name} <_ptype-{name.replace(' ', '_')}`_
```

Also, why replace space with underscore - spaces in links do seem to be allowed.

Reply:
Sorry, I'm looking at the wrong part of the text; I meant use 
```
- _`{name} <ptype-{name.replace(' ', '_')}`
```
which may not in fact work as the only thing I did note is that one can have an in-test target using _`target`

Reply:
I just tried this and a couple variations. I can't get any of them work. I don't think RST has inline link target functionality.

Reply:
It does have it, but quite possibly it does not work in tables -
see https://docs.readthedocs.io/en/stable/guides/cross-referencing-with-sphinx.html
```
For example, an in-line target inside a paragraph:

We can also create _`in-line targets` within an element on your page,
allowing you to, for example, reference text *within* a paragraph.
```

Reply:
I found https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#inline-internal-targets
which has a nice explanation how to do this and other link types.

The link target that works is  ``_`length` `` with reference ``numpydoc_xref_aliases['length'] = "length <length>"

What doesn't work is the MUCH cleaner option `` _`length <ptype-length>` ``  with reference ``numpydoc_xref_aliases['length'] = "length <ptype-length>". Dumping the raw physical type names, unprepended by a descriptive prefix is potentially restrictive elsewhere in the docs. I haven't checked, but I suspect this will cause problems if a section is ever titled "work" or "time" in ANY section of the astropy docs (or docs that use astropy in their intersphinx, depending on the setup).

The alternative that avoids this issue has the somewhat ugly hanging column (see example from opening comment). This is a current limitation of sphinx.

Reply:
I've gone with the former `_`length` ` with reference ``numpydoc_xref_aliases['length'] = "length <'length'>" ``

Reply:
GTG!